version: '3.8'

networks:
  shared-tier:

services:
  grafana:
    image: grafana/grafana:10.1.1
    container_name: grafana
    ports:
      - '3000:3000'
    restart: unless-stopped
    environment:
      GF_SECURITY_ADMIN_USER: admin
      GF_SECURITY_ADMIN_PASSWORD: admin
      GF_RENDERING_SERVER_URL: http://renderer:8081/render
      GF_RENDERING_CALLBACK_URL: http://grafana:3000/
      GF_LOG_FILTERS: rendering:debug
    volumes:
      - ./grafana/provisioning:/etc/grafana/provisioning
      - grafana_data:/var/lib/grafana
    networks:
      - shared-tier
    healthcheck:
        test: curl --fail http://localhost:3000 || exit 1
        interval: 5s
        timeout: 5s
        retries: 5

  renderer:
    image: grafana/grafana-image-renderer:3.9.0
    container_name: renderer
    environment:
      TZ: Europe/Vilnius
      RENDERING_VERBOSE_LOGGING: "true"
      RENDERING_CLUSTERING_TIMEOUT: "30"
      LOG_LEVEL: debug
    ports:
      - '8081:8081'
    restart: unless-stopped
    networks:
      - shared-tier

  prometheus:
    image: prom/prometheus
    container_name: prometheus
    command:
      - '--config.file=/etc/prometheus/prometheus.yaml'
    ports:
      - '9090:9090'
    restart: unless-stopped
    volumes:
      - ./prometheus:/etc/prometheus
      - prometheus_data:/prometheus
    networks:
      - shared-tier

  web:
    image: alpine
    depends_on:
      grafana:
        condition: service_healthy

volumes:
  grafana_data: {}
  prometheus_data: {}
